public/app/features/alerting/unified/rule-editor/formProcessing.ts

```diff
@@ -0,0 +1,150 @@
+      return {
+        ...query,
+        model: {
+          ...query.model,
+          instant: isInstant,
+          range: !isInstant, // we cannot have both instant and range queries in alerting
+        },
+      };
+    }),
+  };
+}
+
+export function areQueriesTransformableToSimpleCondition(
+  dataQueries: Array<AlertQuery<AlertDataQuery | ExpressionQuery>>,
+  expressionQueries: Array<AlertQuery<ExpressionQuery>>
+) {
+  if (dataQueries.length !== 1) {
+    return false;
+  }
+  const singleReduceExpressionInInstantQuery =
+    'instant' in dataQueries[0].model && dataQueries[0].model.instant && expressionQueries.length === 1;
+
+  if (expressionQueries.length !== 2 && !singleReduceExpressionInInstantQuery) {
+    return false;
+  }
+
+  const query = dataQueries[0];
+
+  if (query.refId !== SimpleConditionIdentifier.queryId) {
+    return false;
+  }
+
+  const reduceExpressionIndex = expressionQueries.findIndex(
+    (query) => query.model.type === ExpressionQueryType.reduce && query.refId === SimpleConditionIdentifier.reducerId
+  );
+  const reduceExpression = expressionQueries.at(reduceExpressionIndex);
+  const reduceOk =
+    reduceExpression &&
+    reduceExpressionIndex === 0 &&
+    (reduceExpression.model.settings?.mode === ReducerMode.Strict ||
+      reduceExpression.model.settings?.mode === undefined);
+
+  const thresholdExpressionIndex = expressionQueries.findIndex(
+    (query) =>
+      query.model.type === ExpressionQueryType.threshold && query.refId === SimpleConditionIdentifier.thresholdId
+  );
+  const thresholdExpression = expressionQueries.at(thresholdExpressionIndex);
+  const conditions = thresholdExpression?.model.conditions ?? [];
+  const thresholdIndexOk = singleReduceExpressionInInstantQuery
+    ? thresholdExpressionIndex === 0
+    : thresholdExpressionIndex === 1;
+  const thresholdOk = thresholdExpression && thresholdIndexOk && conditions[0]?.unloadEvaluator === undefined;
+  return (Boolean(reduceOk) || Boolean(singleReduceExpressionInInstantQuery)) && Boolean(thresholdOk);
+}
+
+export function isExpressionQueryInAlert(
+  query: AlertQuery<AlertDataQuery | ExpressionQuery>
+): query is AlertQuery<ExpressionQuery> {
+  return isExpressionQuery(query.model);
+}
+
+export function isAlertQueryOfAlertData(
+  query: AlertQuery<AlertDataQuery | ExpressionQuery>
+): query is AlertQuery<AlertDataQuery> {
+  return !isExpressionQuery(query.model);
+}
+
+// the backend will always execute "hidden" queries, so we have no choice but to remove the property in the front-end
+// to avoid confusion. The query editor shows them as "disabled" and that's a different semantic meaning.
+// furthermore the "AlertingQueryRunner" calls `filterQuery` on each data source and those will skip running queries that are "hidden"."
+// It seems like we have no choice but to act like "hidden" queries don't exist in alerting.
+export const revealHiddenQueries = (ruleDefinition: RuleFormValues): RuleFormValues => {
```

Comment:
I changed the name from `ignoreHiddenQueries` because if I understand correctly, we don't remove any queries, we only remove the `hidden` query parameter

Reply:
💯